### DISCUSSION TITLE
Bank encoders

### DISCUSSION THREAD

**Discussion by pmjcontroller1** [shanteacontrols/OpenDeck#224] [Q&A]
Good morning.  I have a question. I can't find documentation on how to make encoder banks or buttons.  I need to make a bank with 8 encoders I have 4 buttons, each button has to activate 1 bank where each encoder has its CC, when pressing another button it changes the encoders to their CC of this bank

**Comment by stllvdv** [comment]
I was thinking about that as well, but didn't have time to actually look into it. I think all you have to do though is connect your buttons and choose the correct message type for each button in the OpenDeck Configurator, see IMG attached.

<img width="1203" alt="button-preset-change" src="https://github.com/shanteacontrols/OpenDeck/assets/29223372/4214886e-f423-4228-b835-f1977dce4d27">

**Reply by pmjcontroller1** [reply]
I think that this solution is for a button action, in this case the button action would be to change the notes and the channel of the encoders.  the idea is to have four encoders and four buttons the buttons activate bank 1 button 2 activate bank 2 button 3 activate bank 3 button 4 activate bank 4 and in each bank the 4 same physical encoders have different cc

**Reply by stllvdv** [reply]
To my understanding you just need to properly connect and then define 4 buttons (0, 1, 2 and 3 for example) with "message type" defined as "preset change", then define preset 0 for button 0, preset 1 for button 1 and yada yada yada… each button press will update the presets for all encoders accordingly. 

Just make sure you've actually saved 4 different presets (change presets on the global page).
